sno/cli_util.py

```diff
@@ -45,6 +46,21 @@ def startup_load_git_init_config():
         os.environ["GIT_CONFIG_PARAMETERS"] = f"'init.defaultBranch=main'{existing}"
 
 
+def git_remote_environment():
```

Comment:
This needs to be for every git command, since we don't know with partial clones which commands might need to interact with a promisor remote.